include/openthread/tcp.h

```diff
@@ -0,0 +1,639 @@
+ * provided in this file.
+ */
+typedef struct otTcpEndpoint
+{
+    struct otTcpEndpoint *mNext;                      ///< A pointer to the next TCP endpoint (internal use only)
+    void *mContext;                                   ///< A pointer to application-specific context
+    otTcpEstablished mEstablishedCallback;            ///< "Established" callback function
+    otTcpSendDone mSendDoneCallback;                  ///< "Send done" callback function
+    otTcpSendReady mSendReadyCallback;                ///< "Send ready" callback function
+    otTcpReceiveAvailable mReceiveAvailableCallback;  ///< "Receive available" callback function
+    otTcpDisconnected mDisconnectedCallback;          ///< "Disconnected" callback function
+
+    /* Other implementation-defined fields go here. */
+} otTcpEndpoint;
+
+/**
+ * This structure contains arguments to the otTcpEndpointInitialize() function.
+ */
+typedef struct otTcpEndpointInitializeArgs
+{
+    otTcpEstablished mEstablishedCallback;            ///< "Established" callback function
+    otTcpSendDone mSendDoneCallback;                  ///< "Send done" callback function
+    otTcpBytesAcked mBytesAckedCallback;              ///< "Bytes acked" callback
+    otTcpSendReady mSendReadyCallback;                ///< "Send ready" callback function
+    otTcpReceiveAvailable mReceiveAvailableCallback;  ///< "Receive available" callback function
+    otTcpDisconnected mDisconnectedCallback;          ///< "Disconnected" callback function
+    void *mContext;                                   ///< Pointer to application-specific context
+    void *mReceiveBuffer;                             ///< Pointer to memory provided to the system for the TCP receive buffer
+    size_t mReceiveBufferSize;                        ///< Size of memory provided to the system for the TCP receive buffer
+} otTcpEndpointInitializeArgs;
+
+/**
+ * Initializes a TCP endpoint.
+ *
+ * Calling this function causes OpenThread to keep track of the TCP endpoint
+ * and store and retrieve TCP data inside the @p aEndpoint. The application
+ * should refrain from directly accessing or modifying the fields in
+ * @p aEndpoint. If the application needs to reclaimthe memory backing
+ * @p aEndpoint, it should call otTcpEndpointDeinitialize().
+ *
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aEndpoint  A pointer to a TCP endpoint structure.
+ * @param[in]  aArgs      A pointer to a structure of arguments.
+ *
+ * @retval OT_ERROR_NONE    Successfully opened the TCP endpoint.
+ * @retval OT_ERROR_FAILED  Failed to open the TCP endpoint.
+ */
+otError otTcpEndpointInitialize(otInstance *aInstance, otTcpEndpoint *aEndpoint, otTcpEndpointInitializeArgs *aArgs);
+
+/**
+ * Obtains the otInstance that was associated with @p aEndpoint upon
+ * initialization.
+ *
+ * @param[in]  aEndpoint  The TCP endpoint whose instance to obtain.
+ *
+ * @retval  The otInstance pointer associated with @p aEndpoint.
```

Comment:
```suggestion
 * @returns  The otInstance pointer associated with @p aEndpoint.
```

`@retval` requires a return value followed by a discription. For this case `@returns` would be better.
```
@retval <return value> { description }
```